### PR TITLE
Allow not using CPU limits on k8s apps

### PR DESCRIPTION
**Pull request by grosser**
see https://github.com/zendesk/samson/issues/2820 for details
/cc @zendesk/samson @jonmoter @rata 

quick dump of how I think this might wok ... looks alright ?

### CODE REVIEW

**Comment by grosser** [comment]
screen shot 2018-06-26 at 9 01 41 am
screen shot 2018-06-26 at 9 02 02 am

**Comment by rata** [comment]
Awesome! But not sure mem limit might be desirable, at least in some cases. It does not cause such a penalty, just enforces the mem limit.  What do you think?

**Comment by grosser** [comment]
updated!

**Comment by rata** [comment]
@grosser LGTM! :)

File: test/test_helper.rb [hunk 1]
```diff
@@ -215,6 +215,14 @@ def silence_thread_exceptions
   ensure
     Thread.report_on_exception = old
   end
+
+  def stub_const(base, name, value)
```
Comment (grosser) [inline test/test_helper.rb]: fyi @ragurney

**Pull request merged**